src/shared/services/bookmarks-views-plays-service.ts

```diff
@@ -0,0 +1,370 @@
+					this.insertInitialCount(
+						`${action}Init` as EventInitAction,
+						contentType,
+						contentUuid,
+						user,
+						silent
+					);
+				} else {
+					// Bookmark related action failed
+					console.error('Failed to store action into the database', null, {
+						response,
+						query: print(mutation),
+						variables,
+					});
+					toastService.danger(
+						action === 'bookmark'
+							? i18n.t('Het aanmaken van de bladwijzer is mislukt')
+							: i18n.t('Het verwijderen van de bladwijzer is mislukt')
+					);
+				}
+			}
+
+			// Finished incrementing
+		} catch (err) {
+			const error = new CustomError('Failed to store user action to the database', err, {
+				action,
+				contentType,
+				contentUuid,
+				user,
+			});
+			if (silent) {
+				console.error(error);
+			} else {
+				throw error;
+			}
+		}
+	}
+
+	public static async getItemCounts(
+		itemUuid: string,
+		user: Avo.User.User | null
+	): Promise<BookmarkViewPlayCounts> {
+		const response = await dataService.query({
+			query: GET_ITEM_BOOKMARK_VIEW_PLAY_COUNTS,
+			variables: { itemUuid, profileId: get(user, 'profile.id', null) },
+		});
+		const isBookmarked = !!get(response, 'data.app_item_bookmarks[0]');
+		const bookmarkCount = get(response, 'data.app_item_bookmarks_aggregate.aggregate.count', 0);
+		const viewCount = get(response, 'data.app_item_views[0].count', 0);
+		const playCount = get(response, 'data.app_item_plays[0].count', 0);
+		return {
+			bookmarkCount,
+			viewCount,
+			playCount,
+			isBookmarked,
+		};
+	}
+
+	public static async getCollectionCounts(
+		collectionUuid: string,
+		user: Avo.User.User | null
+	): Promise<BookmarkViewPlayCounts> {
+		const response = await dataService.query({
+			query: GET_COLLECTION_BOOKMARK_VIEW_PLAY_COUNTS,
+			variables: { collectionUuid, profileId: get(user, 'profile.id', null) },
+		});
+		const isBookmarked = !!get(response, 'data.app_collection_bookmarks[0]');
+		const bookmarkCount = get(
+			response,
+			'data.app_collection_bookmarks_aggregate.aggregate.count',
+			0
+		);
+		const viewCount = get(response, 'data.app_collection_views[0].count', 0);
+		const playCount = get(response, 'data.app_collection_plays[0].count', 0);
+		return {
+			bookmarkCount,
+			viewCount,
+			playCount,
+			isBookmarked,
+		};
+	}
+
+	/**
+	 * Toggles the bookmark for the provided item or collection or bundle
+	 * @param contentId
+	 * @param user
+	 * @param type
+	 * @param isBookmarked
+	 * @return {boolean} returns true of the operation was successful, otherwise false
+	 */
+	public static async toggleBookmark(
+		contentId: string,
+		user: Avo.User.User,
+		type: EventContentTypeSimplified,
+		isBookmarked: boolean
+	): Promise<boolean> {
+		try {
+			if (!contentId) {
+				throw new CustomError(
+					`Failed to bookmark ${type} because the ${type} doesn't seem to be loaded yet`,
+					null,
+					{ contentId }
+				);
+			}
+			if (isBookmarked) {
```

Comment:
`await BookmarksViewsPlaysService.action(isBookmarked ? 'unbookmark' : 'bookmark', type, contentId, user, false);` kan deze if-statement vervangen.

Reply:
fixed in: feature/bookmarks-in-workspace